--- c-libedit/filecomplete.c.orig	2021-05-22 20:54:26.000000000 +0200
+++ c-libedit/filecomplete.c	2023-03-08 13:10:03.359485491 +0100
@@ -822,7 +822,11 @@
     char **(*attempted_completion_function)(const char *, int, int),
     const wchar_t *word_break, const wchar_t *special_prefixes,
     const char *(*app_func)(const char *), size_t query_items,
-    int *completion_type, int *over, int *point, int *end)
+    int *completion_type, int *over, int *point, int *end,
+    const wchar_t *(*bsd_unused0)(const wchar_t *, const wchar_t *),
+    wchar_t *(*bsd_unused1)(const wchar_t *),
+    char *(*bsd_unused2)(const char *)
+    )
 {
 	return fn_complete2(el, complete_func, attempted_completion_function,
 	    word_break, special_prefixes, app_func, query_items,
@@ -839,5 +843,5 @@
 {
 	return (unsigned char)fn_complete(el, NULL, NULL,
 	    break_chars, NULL, NULL, (size_t)100,
-	    NULL, NULL, NULL, NULL);
+	    NULL, NULL, NULL, NULL, NULL, NULL, NULL);
 }
--- c-libedit/filecomplete.h.orig	2023-03-08 13:00:52.914532658 +0100
+++ c-libedit/filecomplete.h	2023-03-08 13:10:51.307453982 +0100
@@ -35,7 +35,11 @@
     char *(*)(const char *, int),
     char **(*)(const char *, int, int),
     const wchar_t *, const wchar_t *, const char *(*)(const char *), size_t,
-    int *, int *, int *, int *);
+    int *, int *, int *, int *,
+    const wchar_t *(*)(const wchar_t *, const wchar_t *),
+    wchar_t *(*)(const wchar_t *),
+    char *(*)(const char *)
+    );
 int fn_complete2(EditLine *,
     char *(*)(const char *, int),
     char **(*)(const char *, int, int),
--- c-libedit/readline.c.orig	2023-03-08 13:00:28.000000000 +0100
+++ c-libedit/readline.c	2023-03-08 13:00:52.914532658 +0100
@@ -1961,7 +1961,7 @@
 	    _rl_completion_append_character_function,
 	    (size_t)rl_completion_query_items,
 	    &rl_completion_type, &rl_attempted_completion_over,
-	    &rl_point, &rl_end);
+	    &rl_point, &rl_end, NULL, NULL, NULL);
 
 
 }
